pkg/apis/v1alpha1/common_validations.go

```diff
@@ -77,3 +90,26 @@ func validPath(path string, validPaths map[string]bool, validPrefixes []string)
 
 	return false
 }
+
+func validateSelectors(selectors Selectors, validPaths map[string]bool, validPrefixes []string) error {
+	var err error
+
+	labelSelector := &metav1.LabelSelector{
+		MatchLabels:      selectors.Selector,
+		MatchExpressions: selectors.SelectorMatchExpressions,
+	}
+
+	_, err = metav1.LabelSelectorAsSelector(labelSelector)
+	if err != nil {
+		return fmt.Errorf("selectorMatchExpressions or selectors are not valid: %w", err)
```

Comment:
uh I think this can be deceiving in case the validation fails for `selector` but _not_ `selectorMatchExpressions`, e.g.:

```yaml
apiVersion: carto.run/v1alpha1
kind: ClusterSupplyChain
metadata:
  name: basic-image-to-url
spec:
  selector:
    foo-: web

  selectorMatchFields:
    - key: spec.image
      operator: Exists
```

has a problem in `selector` but the result from the webhook is:

```
kapp: Error: Applying create clustersupplychain/basic-image-to-url (carto.run/v1alpha1) cluster:
  Creating resource clustersupplychain/basic-image-to-url (carto.run/v1alpha1) cluster:
    API server says: admission webhook "supply-chain-validator.cartographer.com" denied the request: error validating clustersupplychain [basic-image-to-url]: 

 \/ \/ \/ \/ \/ \/ \/
selectorMatchExpressions or selectors are not valid: key:
      Invalid value: "foo-": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]') (reason: error validating clustersupplychain [basic-image-to-url]: selectorMatchExpressions or selectors are not valid: key: Invalid value: "foo-": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]'))

```

Reply:
Ya, we chatted about that. We can make the call twice for a better message